lxd/device/disk.go

```diff
@@ -345,6 +345,43 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 		}
 	}
 
+	// Extract initial configuration from the profile and validate them against appropriate storage driver.
+	// Currently initial configuration is only applicable to root disk devices.
+
+	initialConfig := make(map[string]string)
+	for k, v := range d.config {
+		if strings.Contains(k, "initial.") {
+			newKey := strings.TrimPrefix(k, "initial.")
+			initialConfig[newKey] = v
+		}
+	}
+
+	if len(initialConfig) > 0 {
+		if !shared.IsRootDiskDevice(d.config) {
+			return fmt.Errorf("Non-root disk device cannot contain initial.* configuration")
+		}
+
+		d.pool, err = storagePools.LoadByName(d.state, d.config["pool"])
```

Comment:
Can we avoid loading this twice and move it within the section above that has the comment:

```
// Only perform expensive instance pool volume checks when not validating a profile and after
// device expansion has occurred (to avoid doing it twice during instance load).
```